src/events/http.js

```diff
@@ -305,14 +305,11 @@ function httpWrapper(wrappedFunction) {
                     }
                 });
 
-                clientRequest.on('close', () => {
-                    resolveHttpPromise(httpEvent, resolve, startTime);
-                });
-
                 clientRequest.on('response', (res) => {
-                    res.on('data', (chunk) => { data += chunk; });
+                    res.on('data', (chunk) => { chunks.push(chunk); });
                     res.on('end', () => {
-                        setJsonPayload(httpEvent, 'response_body', data);
+                        setJsonPayload(httpEvent, 'response_body', Buffer.concat(chunks), res.headers['content-encoding']);
```

Comment:
what happens if the response doesn't have a `content-encoding` header?

Reply:
It will be `undefined` which means that the code will not try to decode it.